Solution1/Solution1.Tests/BL/Services/WeatherApiServiceTests.cs

```diff
@@ -110,9 +109,37 @@ public async Task GetForecastByCityNameAsync_ReturnedForecastWeatherApiDTO_Succe
             Assert.True(new CompareLogic().Compare(expected, result).AreEqual);
         }
 
-        private void SetHttpHandlerSettings(Mock<HttpMessageHandler> httpMessageHandler, HttpResponseMessage response, string uri)
+        [Fact]
+        public async Task GetByCityNameAsync_GenerateTaskCanceledException_Success()
+        {
+            // Arrange
+            var cancellationTokenSource = new CancellationTokenSource();
+            cancellationTokenSource.Cancel();
+            SetHttpHandlerSettingsForTaskCanceledException();
+
+            // Act
+            
+            // Assert
+            await Assert.ThrowsAsync<TaskCanceledException>(async() => await _weatherApiService.GetByCityNameAsync(cityName, cancellationTokenSource.Token));
```

Comment:
FYI: its both Act / Assert in one command

Reply:
+